internal/configs/version2/__snapshots__/templates_test.snap

```diff
@@ -0,0 +1,6127 @@
+        add_header grpc-message 'resource exhausted';
+        return 204;
+    }
+
+    location @grpc_unimplemented {
+        default_type application/grpc;
+        add_header content-type application/grpc;
+        add_header grpc-status 12;
+        add_header grpc-message unimplemented;
+        return 204;
+    }
+
+    location @grpc_internal {
+        default_type application/grpc;
+        add_header content-type application/grpc;
+        add_header grpc-status 13;
+        add_header grpc-message 'internal error';
+        return 204;
+    }
+
+    location @grpc_unavailable {
+        default_type application/grpc;
+        add_header content-type application/grpc;
+        add_header grpc-status 14;
+        add_header grpc-message unavailable;
+        return 204;
+    }
+
+    location @grpc_unauthenticated {
+        default_type application/grpc;
+        add_header content-type application/grpc;
+        add_header grpc-status 16;
+        add_header grpc-message unauthenticated;
+        return 204;
+    }
+
+        
+    
+}
+
+---
+
+[TestExecuteVirtualServerTemplate_RendersTemplateWithServerGunzipNotSet - 1]
+
+upstream test-upstream {
+    zone test-upstream 256k;
+    random;
+    server 10.0.0.20:8001 max_fails=4 fail_timeout=10s slow_start=10s max_conns=31;
+    keepalive 32;
+    queue 10 timeout=60s;
+    sticky cookie test expires=25s path=/tea;
+
+    ntlm;
+}
+
+upstream coffee-v1 {
+    zone coffee-v1 256k;
+    server 10.0.0.31:8001 max_fails=8 fail_timeout=15s max_conns=2;
+
+    
+}
+
+upstream coffee-v2 {
+    zone coffee-v2 256k;
+    server 10.0.0.32:8001 max_fails=12 fail_timeout=20s max_conns=4;
+
+    
+}
+
+split_clients $request_id $split_0 {
+    50% @loc0;
+    50% @loc1;
+}
+map $match_0_0 $match {
+    ~^1 @match_loc_0;
+    default @match_loc_default;
+}
+map $http_x_version $match_0_0 {
+    v2 1;
+    default 0;
+}
+# HTTP snippet
+limit_req_zone $url zone=pol_rl_test_test_test:10m rate=10r/s;
+
+server {
+    listen 80 proxy_protocol;
+    listen [::]:80 proxy_protocol;
+
+
+    server_name example.com;
+    status_zone example.com;
+    set $resource_type "virtualserver";
+    set $resource_name "";
+    set $resource_namespace "";
+    listen 443 ssl proxy_protocol;
+    listen [::]:443 ssl proxy_protocol;
+
+    http2 on;
+    ssl_certificate cafe-secret.pem;
+    ssl_certificate_key cafe-secret.pem;
+    ssl_client_certificate ingress-mtls-secret;
+    ssl_verify_client on;
+    ssl_verify_depth 2;
+    if ($scheme = 'http') {
+        return 301 https://$host$request_uri;
+    }
+
+    server_tokens "off";
+    set_real_ip_from 0.0.0.0/0;
+    real_ip_header X-Real-IP;
+    real_ip_recursive on;
+    allow 127.0.0.1;
+    deny all;
+    deny 127.0.0.1;
+    allow all;
+    limit_req_log_level error;
+    limit_req_status 503;
+    limit_req zone=pol_rl_test_test_test burst=5
+         delay=10;
+    auth_jwt "My Api";
+    auth_jwt_key_file jwk-secret;
+    app_protect_enable ;
```

Comment:
![image](https://github.com/nginxinc/kubernetes-ingress/assets/128620221/3c219b94-df19-4b95-8f26-9c30bc3c9bad)
should this say
```
app_protect_enable on;
```